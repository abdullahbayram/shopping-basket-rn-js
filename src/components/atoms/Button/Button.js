import * as React from 'react';
import PropTypes from 'prop-types';
import { Button as PaperButton } from 'react-native-paper';

const Button = ({ children, onPress, icon = '', mode = 'contained' }) => (
  <PaperButton onPress={onPress} icon={icon} mode={mode}>
    {children}
  </PaperButton>
);

Button.propTypes = {
  children: PropTypes.node.isRequired,
  onPress: PropTypes.func.isRequired,
  icon: PropTypes.string,
  mode: PropTypes.string,
};

export default Button;
