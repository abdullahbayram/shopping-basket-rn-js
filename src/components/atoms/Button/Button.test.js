import React from 'react';
import { render, fireEvent, screen } from '@testing-library/react-native';
import Button from '.';

const BUTTON_TEXT_CLICK_ME = 'Click Me';
const BUTTON_TEXT_PRESS_ME = 'Press Me';

beforeEach(() => {
  jest.clearAllMocks(); // Clears all mocks in the test suite
});

const mockOnPress = jest.fn();

const renderButton = (onPress, mode, children, icon, disabled) => {
  return render(
    <Button onPress={onPress} icon={icon} mode={mode} disabled={disabled}>
      {children}
    </Button>,
  );
};

describe('<Button />', () => {
  it('matches the snapshot', () => {
    const { toJSON } = renderButton(() => {}, 'outlined', BUTTON_TEXT_CLICK_ME, 'camera');
    expect(toJSON()).toMatchSnapshot();
  });
  it('matches the snapshot(defaultProps)', () => {
    const { toJSON } = renderButton(() => {}, undefined, BUTTON_TEXT_CLICK_ME);
    expect(toJSON()).toMatchSnapshot();
  });
  it('renders correctly with children and props', () => {
    renderButton(() => {}, 'outlined', BUTTON_TEXT_CLICK_ME, 'camera');
    expect(screen.getByText(BUTTON_TEXT_CLICK_ME)).toBeTruthy();
  });

  it('renders correctly with default props', () => {
    renderButton(mockOnPress, undefined, BUTTON_TEXT_PRESS_ME);
    const button = screen.getByText(BUTTON_TEXT_PRESS_ME);
    expect(button).toBeTruthy();
    fireEvent.press(button);
    expect(mockOnPress).toHaveBeenCalledTimes(1);
  });

  it('calls onPress when the button is pressed', () => {
    renderButton(mockOnPress, undefined, BUTTON_TEXT_PRESS_ME);
    const button = screen.getByText(BUTTON_TEXT_PRESS_ME);
    fireEvent.press(button);
    expect(mockOnPress).toHaveBeenCalledTimes(1);
  });

  it('renders correctly with a custom icon and mode', () => {
    renderButton(mockOnPress, 'outlined', BUTTON_TEXT_PRESS_ME, 'camera');
    const button = screen.getByText(BUTTON_TEXT_PRESS_ME);
    expect(button).toBeTruthy();
    fireEvent.press(button);
    expect(mockOnPress).toHaveBeenCalledTimes(1);
  });

  it('does not call onPress when the button is disabled', () => {
    renderButton(mockOnPress, undefined, BUTTON_TEXT_PRESS_ME, 'camera', true);
    const button = screen.getByText(BUTTON_TEXT_PRESS_ME);
    fireEvent.press(button);
    expect(mockOnPress).not.toHaveBeenCalled();
  });
});
