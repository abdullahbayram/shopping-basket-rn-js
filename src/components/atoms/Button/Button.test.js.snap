// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> matches the snapshot with custom props 1`] = `
<View
  collapsable={false}
  style={
    {
      "backgroundColor": "rgba(98, 91, 113, 1)",
      "borderRadius": 20,
      "shadowColor": "#000",
      "shadowOffset": {
        "height": 0,
        "width": 0,
      },
      "shadowOpacity": 0,
      "shadowRadius": 0,
    }
  }
  testID="button-container-outer-layer"
>
  <View
    collapsable={false}
    style={
      {
        "backgroundColor": "rgba(98, 91, 113, 1)",
        "borderColor": "rgba(121, 116, 126, 1)",
        "borderRadius": 20,
        "borderStyle": "solid",
        "borderWidth": 1,
        "flex": undefined,
        "minWidth": 64,
        "shadowColor": "#000",
        "shadowOffset": {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 0,
        "shadowRadius": 0,
      }
    }
    testID="button-container"
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": false,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        [
          {
            "overflow": "hidden",
          },
          {
            "borderRadius": 20,
          },
        ]
      }
      testID="button"
    >
      <View
        style={
          [
            {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
            },
            undefined,
          ]
        }
      >
        <View
          style={
            [
              {
                "marginLeft": 12,
                "marginRight": -4,
              },
              {
                "marginLeft": 16,
                "marginRight": -16,
              },
              false,
            ]
          }
          testID="button-icon-container"
        >
          <Text />
        </View>
        <Text
          numberOfLines={1}
          selectable={false}
          style={
            [
              {
                "textAlign": "left",
              },
              {
                "color": "rgba(28, 27, 31, 1)",
                "writingDirection": "ltr",
              },
              [
                {
                  "fontFamily": "System",
                  "fontSize": 14,
                  "fontWeight": "500",
                  "letterSpacing": 0.1,
                  "lineHeight": 20,
                },
                [
                  {
                    "marginHorizontal": 16,
                    "marginVertical": 9,
                    "textAlign": "center",
                  },
                  false,
                  {
                    "marginHorizontal": 24,
                    "marginVertical": 10,
                  },
                  undefined,
                  false,
                  {
                    "color": "rgba(125, 82, 96, 1)",
                    "fontFamily": "System",
                    "fontSize": 14,
                    "fontWeight": "500",
                    "letterSpacing": 0.1,
                    "lineHeight": 20,
                  },
                  undefined,
                ],
              ],
            ]
          }
          testID="button-text"
        >
          Click Me
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`<Button /> matches the snapshot with default props 1`] = `
<View
  collapsable={false}
  style={
    {
      "backgroundColor": "rgba(98, 91, 113, 1)",
      "borderRadius": 20,
      "shadowColor": "#000",
      "shadowOffset": {
        "height": 0,
        "width": 0,
      },
      "shadowOpacity": 0,
      "shadowRadius": 0,
    }
  }
  testID="button-container-outer-layer"
>
  <View
    collapsable={false}
    style={
      {
        "backgroundColor": "rgba(98, 91, 113, 1)",
        "borderColor": "rgba(121, 116, 126, 1)",
        "borderRadius": 20,
        "borderStyle": "solid",
        "borderWidth": 1,
        "flex": undefined,
        "minWidth": 64,
        "shadowColor": "#000",
        "shadowOffset": {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 0,
        "shadowRadius": 0,
      }
    }
    testID="button-container"
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": false,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        [
          {
            "overflow": "hidden",
          },
          {
            "borderRadius": 20,
          },
        ]
      }
      testID="button"
    >
      <View
        style={
          [
            {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
            },
            undefined,
          ]
        }
      >
        <Text
          numberOfLines={1}
          selectable={false}
          style={
            [
              {
                "textAlign": "left",
              },
              {
                "color": "rgba(28, 27, 31, 1)",
                "writingDirection": "ltr",
              },
              [
                {
                  "fontFamily": "System",
                  "fontSize": 14,
                  "fontWeight": "500",
                  "letterSpacing": 0.1,
                  "lineHeight": 20,
                },
                [
                  {
                    "marginHorizontal": 16,
                    "marginVertical": 9,
                    "textAlign": "center",
                  },
                  false,
                  {
                    "marginHorizontal": 24,
                    "marginVertical": 10,
                  },
                  undefined,
                  false,
                  {
                    "color": "rgba(125, 82, 96, 1)",
                    "fontFamily": "System",
                    "fontSize": 14,
                    "fontWeight": "500",
                    "letterSpacing": 0.1,
                    "lineHeight": 20,
                  },
                  undefined,
                ],
              ],
            ]
          }
          testID="button-text"
        >
          Click Me
        </Text>
      </View>
    </View>
  </View>
</View>
`;
